src/app/workflow/dag/cwl-viewer/cwl-viewer.component.ts

```diff
@@ -27,13 +31,11 @@ import { Subject } from 'rxjs';
   styleUrls: ['./cwl-viewer.scss']
 })
 
-export class CwlViewerComponent implements OnInit, OnDestroy {
+export class CwlViewerComponent implements OnInit, OnChanges, OnDestroy {
 
-  @Input() workflow: ExtendedWorkflow;
   @Input() set selectedVersion(value: WorkflowVersion) {
```

Comment:
The only reason to have this setter was to invoke `this.onChange()`. Now that you've made the code better by just using `ngOnChanges`, I think you should just be able to change this to `@Input() version: WorkflowVersion`.

Reply:
reminder, version is only set there it seems.  if you remove the setter, you might as well remove the version variable altogether and use the selectedVersion instead.

Reply:
Sorry, yes I meant `@Input selectedVersion: WorkflowVersion`, which then means as Gary says, that your code should reference this.selectedVersion.

Reply:
Hm, actually I realize now that the image won't appear if the setter is removed.

Reply:
I think the version stays null and so the ngOnChanges part doesn't never goes because of the if check